routing/router.go

```diff
@@ -2394,15 +2396,28 @@ func (r *ChannelRouter) extractChannelUpdate(
 		update = &onionErr.Update
 	case *lnwire.FailFeeInsufficient:
 		update = &onionErr.Update
+
+		if onionErr.IncomingUpdate != nil {
+			updates = append(updates, onionErr.IncomingUpdate)
+		}
 	case *lnwire.FailIncorrectCltvExpiry:
 		update = &onionErr.Update
 	case *lnwire.FailChannelDisabled:
 		update = &onionErr.Update
 	case *lnwire.FailTemporaryChannelFailure:
+		if onionErr.Update == nil {
```

Comment:
Curious, why was this added?

Reply:
IIRC, without this a `nil` update would be appended to the new `updates` list.